unison-core/src/Unison/Name.hs

```diff
@@ -89,9 +100,13 @@ fromVar = unsafeFromText . Var.name
 toString :: Name -> String
 toString = Text.unpack . toText
 
-isPrefixOf :: Name -> Name -> Bool
+isPrefixOf :: HasCallStack => Name -> Name -> Bool
 Name a `isPrefixOf` Name b = a `Text.isPrefixOf` b
-Name' a `isPrefixOf` Name' b = toList a `NonEmpty.isPrefixOf` b
+n1@(Name' pa a) `isPrefixOf` n2@(Name' pb b) =
+  if pa == pb then
+    toList a `NonEmpty.isPrefixOf` b
+  else
+    error ("isPrefixOf (" ++ show n1 ++ ") (" ++ show n2 ++ ")")
```

Comment:
Note that the original implementation would just return false if `pa` and `pb` mismatched.

Reply:
Yeah, good point. I think this was a defensive `error` to try to rule out bad calls that mix name types. Want to just return `False` here, or leave the error for investigating later?

Reply:
Oh, I see.  I think looking at where `isPrefixOf` is called (period) will be more helpful than seeing if the error gets thrown.

Reply:
So, I dunno; what do you think?

Reply:
Yeah, I agree investigating calls to `isPrefixOf` is the move here. The `error`s are not that easy to trip at the moment anyway, probably - I'm not trying to actually use `ucm` yet on this patch :P

They're more like reminders that something is off with this API, to be refactored away soon. So I'm leaning towards leaving it in for this patch.

Reply:
Ok sounds good to me.